src/common/containers/Header.jsx

```diff
@@ -50,17 +60,35 @@ export default class Header extends Component {
                         </li>
                     </ul>
                 )}
-
+                    
                 <ul className="order-1 order-md-3 navbar-nav navbar-no-expand ml-auto">
                     <li className="nav-item">
                         <a href="/dashboard" className="nav-link">Dashboard</a>
                     </li>
                     <li className="nav-item">
                         <a href="/blog" className="nav-link">Blog</a>
                     </li>
-                    <li className="nav-item">
-                        <a className="nav-link">{this.state.user.username}</a>
+                    {this.state.userIsAuth?(
+                        <li className="nav-item">
+                            <a href="/singin" className="nav-link">Sing in</a>
+                        </li>       
+                    )
+                    :(
+                    <li className="nav-item dropdown user-menu show" onClick={this.showDropdown}>
```

Comment:
I think opening and closing the drop-down menu can be implemented without using the react state. AdminLTE3 should implement this functionality, check the documentation.